File: work_1/content/code.py
import random

# 生成五个随机数（unsigned int32 范围：0 ~ 2^32 - 1）
random_numbers = [random.randint(0, 2**32 - 1) for _ in range(5)]

# 写入 input.txt 并在终端打印
with open('input.txt', 'w') as f:
    for num in random_numbers:
        f.write(f"{num} ")
        print(num, end=" ")
    print()  # 换行

# 从 input.txt 读取数据并计算均值
with open('input.txt', 'r') as f:
    numbers = list(map(int, f.read().split()))
    mean = sum(numbers) / len(numbers)

# 写入 output.txt 并在终端打印均值
with open('output.txt', 'w') as f:
    f.write(f"{mean:.6f}")  # 保留 6 位小数，可根据需求调整
    print(f"均值为: {mean:.6f}")

File: work_1/content/code.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <stdint.h>

int main() {
    srand((unsigned int)time(NULL));
    const int count = 5; // 固定数量，避免除以零
    unsigned int nums[count];

    // 写入 input.txt 并终端打印
    FILE *input = fopen("input.txt", "w");
    if (!input) { perror("Write file error"); return 1; }
    for (int i = 0; i < count; i++) {
        nums[i] = rand() % (UINT32_MAX + 1);
        fprintf(input, "%u ", nums[i]);
        printf("%u ", nums[i]);
    }
    printf("\n");
    fclose(input);

    // 读取计算均值
    input = fopen("input.txt", "r");
    if (!input) { perror("Read file error"); return 1; }
    unsigned long long sum = 0;
    for (int i = 0; i < count; i++) {
        unsigned int num;
        fscanf(input, "%u", &num);
        sum += num;
    }
    fclose(input);
    float avg = (float)sum / count; // count 固定为5，无除零风险

    // 写入 output.txt 并终端打印
    FILE *output = fopen("output.txt", "w");
    if (!output) { perror("Write output error"); return 1; }
    fprintf(output, "%.2f\n", avg);
    printf("Average: %.2f\n", avg);
    fclose(output);

    return 0;
}